tests/util/test_rwlock.py

```diff
@@ -32,76 +34,120 @@ def _assert_called_before_not_after(self, lst, first_false):
 
     def test_rwlock(self):
         rwlock = ReadWriteLock()
+        key = "key"
+
+        def start_reader_or_writer(
```

Comment:
it's a bit confusing that this is slightly different to the class-level `_start_reader_or_writer`. Can we combine them, even if it means returning three deferreds rather than two?

Reply:
Yep, let's combine them.